ios-xcframework/build.sh

```diff
@@ -97,32 +97,49 @@ do
     -output "$XCFRAMEWORKS_DIR/$CURRENT_FRAMEWORK_NAME.xcframework"
 done
 
-log 'compression' 'Zipping Gutenberg XCFrameworks'
-ZIP_PATH=$XCFRAMEWORKS_DIR/$MAIN_FRAMEWORK_NAME.zip
-zip -rq "$ZIP_PATH" \
-  $XCFRAMEWORKS_DIR/Aztec.xcframework \
-  $XCFRAMEWORKS_DIR/Gutenberg.xcframework \
-  $XCFRAMEWORKS_DIR/React.xcframework \
-  $XCFRAMEWORKS_DIR/RNTAztecView.xcframework \
-  $XCFRAMEWORKS_DIR/yoga.xcframework
-echo "Gutenberg XCFrameworks ZIP generated at $ZIP_PATH"
-
-# In parallel to the project to ship Gutenberg as an XCFramework we are also
-# experimenting with adding React Native into other apps with an XCFramework.
+log 'bulb' 'Work around "Gutenberg class ahead of Gutenberg module" issue'
+# We generate a Gutenberg framework with a Gutenberg class in it.
+# This results in some types in the swiftinterface being defined as Gutenberg.Gutenberg.Type which cannot be resolved.
+# There is no fix, but we can work around it by post-processing the interface and flattening the namespace where appropriate.
 #
-# For convenience, we produce a ZIP with all the React Native XCFrameworks
-# here, even though it would be more efficient to have a dedicated project that
-# mirrors React Native and builds new one whenever a new version is released.
+# See https://developer.apple.com/forums/thread/123253
 #
-# In this current implementation, the approach is wasteful because we are
-# publishing the same ZIP every time. A possible imporvement would be to track
-# the RN version and check against the storage medium to see if an archive for
-# that version is already available.
-log 'compression' 'Zipping React XCFrameworks'
-ZIP_PATH=$XCFRAMEWORKS_DIR/ReactNative.zip
-zip -rq "$ZIP_PATH" $XCFRAMEWORKS_DIR -x \
-  $XCFRAMEWORKS_DIR/Aztec.xcframework \
-  $XCFRAMEWORKS_DIR/Gutenberg.xcframework \
-  $XCFRAMEWORKS_DIR/RNTAztecView.xcframework \
-  $XCFRAMEWORKS_DIR/Pods_Gutenberg.xcframework
-echo "React Native XCFrameworks ZIP generated at $ZIP_PATH"
+# TODO: There has to be a RegEx that can do what this sequence does in one go. Something along the lines of s/(^|\s|\[)Gutenberg\.(?!self)/\1/g but that doesn't throw errors
```

Comment:
Here is an answer I got from ChatGPT: https://regex101.com/r/W38ky9/1. But I'm not sure if it'd cover all your cases...

An alternative could be using https://github.com/apple/swift-syntax to parse and update the `.swiftinterface` file.

Reply:
Thanks! I'll keep that for a followup PR I that's okay.

Reply:
Tracked at internal ref paaHJt-4WG-p2#comment-7616